base_kanban_stage/models/base_kanban_abstract.py

```diff
@@ -0,0 +1,112 @@
+        help='The Kanban stage that this record is currently in',
+        default=lambda s: s._default_stage_id(),
+        domain=lambda s: [('res_model.model', '=', s._name)],
+    )
+    kanban_user_id = fields.Many2one(
+        string='Assigned To',
+        comodel_name='res.users',
+        index=True,
+        track_visibility='onchange',
+        help='User that the record is currently assigned to',
+    )
+    kanban_color = fields.Integer(
+        string='Color Index',
+        help='Color index to be used for the record\'s Kanban card',
+    )
+    kanban_legend_priority = fields.Text(
+        string='Priority Explanation',
+        related='stage_id.legend_priority',
+        help='Explanation text to help users understand how the priority/star'
+             ' mechanism applies to this record (depends on current stage)',
+    )
+    kanban_legend_blocked = fields.Text(
+        string='Special Handling Explanation',
+        related='stage_id.legend_blocked',
+        help='Explanation text to help users understand how the special'
+             ' handling status applies to this record (depends on current'
+             ' stage)',
+    )
+    kanban_legend_done = fields.Text(
+        string='Ready Explanation',
+        related='stage_id.legend_done',
+        help='Explanation text to help users understand how the ready'
+             ' status applies to this record (depends on current stage)',
+    )
+    kanban_legend_normal = fields.Text(
+        string='Normal Handling Explanation',
+        related='stage_id.legend_normal',
+        help='Explanation text to help users understand how the normal'
+             ' handling status applies to this record (depends on current'
+             ' stage)',
+    )
+    kanban_status = fields.Selection(
+        selection=[
+            ('normal', 'Normal Handling'),
+            ('done', 'Ready'),
+            ('blocked', 'Special Handling'),
+        ],
+        string='Kanban Status',
+        default='normal',
+        track_visibility='onchange',
+        required=True,
+        copy=False,
+        help='A record can have one of several Kanban statuses, which are used'
+             ' to indicate whether there are any special situations affecting'
+             ' it. The exact meaning of each status is allowed to vary based'
+             ' on the stage the record is in but they are roughly as follow:\n'
+             '* Normal Handling: Default status, no special situations\n'
+             '* Ready: Ready to transition to the next stage\n'
+             '* Special Handling: Blocked in some way (e.g. must be handled by'
+             ' a specific user)\n'
+    )
+
+    @api.model
+    def _default_stage_id(self):
+        return self.env['base.kanban.stage']
+
+    @api.multi
+    def _read_group_stage_ids(
+        self, domain=None, read_group_order=None, access_rights_uid=None
+    ):
+        stage_model = self.env['base.kanban.stage'].sudo(access_rights_uid)
```

Comment:
If `access_rights_uid` is an optional argument, we should only use sudo if it is provided.